Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm

```diff
@@ -4343,31 +4343,24 @@ static NSDragOperation kit(std::optional<WebCore::DragOperation> dragOperation)
         // guaranteed that the count of UTIs equals the count of files, since some clients only write
         // unique UTIs.
         NSArray *files = [draggingInfo.draggingPasteboard propertyListForType:WebCore::legacyFilesPromisePasteboardType()];
-        if (![files isKindOfClass:[NSArray class]]) {
-            delete dragData;
+        if (![files isKindOfClass:[NSArray class]])
             return false;
-        }
 
         NSString *dropDestinationPath = FileSystem::createTemporaryDirectory(@"WebKitDropDestination");
-        if (!dropDestinationPath) {
-            delete dragData;
+        if (!dropDestinationPath)
             return false;
-        }
 
         size_t fileCount = files.count;
-        Vector<String> *fileNames = new Vector<String>;
+        auto fileNames = Box<Vector<String>>::create();
         NSURL *dropDestination = [NSURL fileURLWithPath:dropDestinationPath isDirectory:YES];
         String pasteboardName = draggingInfo.draggingPasteboard.name;
-        [draggingInfo enumerateDraggingItemsWithOptions:0 forView:m_view.getAutoreleased() classes:@[[NSFilePromiseReceiver class]] searchOptions:@{ } usingBlock:^(NSDraggingItem *draggingItem, NSInteger idx, BOOL *stop) {
-            NSFilePromiseReceiver *item = draggingItem.item;
-            NSDictionary *options = @{ };
-
-            RetainPtr<NSOperationQueue> queue = adoptNS([NSOperationQueue new]);
-            [item receivePromisedFilesAtDestination:dropDestination options:options operationQueue:queue.get() reader:^(NSURL *fileURL, NSError *errorOrNil) {
+        [draggingInfo enumerateDraggingItemsWithOptions:0 forView:m_view.getAutoreleased() classes:@[[NSFilePromiseReceiver class]] searchOptions:@{ } usingBlock:[&](NSDraggingItem *draggingItem, NSInteger idx, BOOL *stop) {
```

Comment:
Ah, I actually left that alone because of:
```
ERROR: Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:4357:  Extra space in capture list.  [whitespace/brackets] [4]
Total errors found: 1 in 1 files
```
...but I'll fix it here, and ignore the style checker false positive.

Thanks for the review!